sdk/edgeorder/Azure.ResourceManager.EdgeOrder/src/Customizations/Models/SiteKey.TokenCredential.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core;
+using Azure.Identity;
+using System.Text.Json;
+using System.Security.Cryptography.X509Certificates;
+using System.Runtime.ConstrainedExecution;
+using System.Net.Sockets;
+using Microsoft.Azure.Management.ResourceManager.Fluent;
+
+namespace Azure.ResourceManager.EdgeOrder.Customizations.Models
+{
+    internal static partial class SiteKeyContent
+    {
+
+        internal static TokenCredential GenerateTokenCredential(this SiteKey siteKey)
+        {
+            var options = new TokenCredentialOptions
+            {
+                AuthorityHost = new Uri(siteKey.AadEndpoint),
+            };
+            return new ClientCertificateCredential(tenantId: siteKey.TenantId,
+                clientId: siteKey.ClientId,
```

Comment:
we will have to honour subject name validation else this will break

Reply:
done by using chained certificate option.